tests/Meziantou.Analyzer.Test/Rules/DoNotUseElseWhenIfBlockJumpsAnalyzerTests.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System.Threading.Tasks;
+using Meziantou.Analyzer.Rules;
+using TestHelper;
+using Xunit;
+
+namespace Meziantou.Analyzer.Test.Rules
+{
+    public sealed class DoNotUseElseWhenIfBlockJumpsAnalyzerTests
+    {
+        private const string IfBreakElse = @"
+class TestClass
+{
+    void Test()
+    {
+        var value = -1;
+        while (true)
+        {
+            if (value < 0)
+            {
+                Incr(ref value);
+                break;
+                void Incr(ref int val) => val++;
+            }
+        [||]else
+            {
+                value--;
+            }
+        }
+    }
+}";
+        private const string IfContinueElse = @"
+class TestClass
+{
+    void Test()
+    {
+        var value = -1;
+        while (true)
+        {
+            if (value < 0)
+            {
+                Incr(ref value);
+                continue;
+                void Incr(ref int val) => val++;
+            }
+        [||]else
+            {
+                value--;
+            }
+        }
+    }
+}";
+        private const string IfGotoElse = @"
+class TestClass
+{
+    void Test()
+    {
+        var value = -1;
+        while (true)
+        {
+            if (value < 0)
+            {
+                Incr(ref value);
+                goto OUT;
+                void Incr(ref int val) => val++;
+            }
+        [||]else
+            {
+                value--;
+            }
+        }
+    OUT:
+        value--;
+    }
+}";
+        private const string IfReturnElse = @"
+class TestClass
+{
+    void Test()
+    {
+        var value = -1;
+        while (true)
+        {
+            if (value < 0)
+            {
+                return;
+            }
+        [||]else
+            {
+                value--;
+            }
+        }
+    }
+}";
+        private const string IfThrowElse = @"
+class TestClass
+{
+    void Test()
+    {
+        var value = -1;
+        while (true)
+        {
+            if (value < 0)
+                throw new System.ArgumentNullException(nameof(value));
+        [||]else
+                value--;
+        }
+    }
+}";
+        private const string IfYieldElse = @"
+class TestClass
+{
+    System.Collections.Generic.IEnumerable<int> Test()
+    {
+        int value = -1;
+        while (true)
+        {
+            if (value < 0)
+            {
+                value++;
+                yield return value;
+            }
+        [||]else
```

Comment:
Removing this else would change the behavior of the code. `yield return` doesn't exit a method. You can test the following method to confirm it:

````c#
IEnumerable<int> A()
{
    yield return 0;
    yield return 1;
    yield return 2;
}
````

You can keep this test but remove the `[||]` and add a new test with `yield break`

Reply:
You're right, I was thinking of `yield break`. I made the change and added a test with `yield return` + a few more tests.